src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+        ? $location`The Briny Deeps`
+        : determineDraggableZoneAndEnsureAccess();
+
+      isEmbezzler ? embezzlerPrep() : freeFightPrep();
+      adventureMacroAuto(targetLocation, Macro.basicCombat());
+      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+    },
+  },
+  {
+    name: "Guaranteed Kramco",
+    available: () => sober() && kramcoGuaranteed(),
+    perform: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return !kramcoGuaranteed();
+    },
+  },
+  {
+    name: "Void Monster",
+    available: () =>
+      sober() &&
+      have($item`cursed magnifying glass`) &&
+      get("cursedMagnifyingGlassCount") === 13 &&
+      get("_voidFreeFights") < 5,
+    perform: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`cursed magnifying glass` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("cursedMagnifyingGlassCount") === 0;
+    },
+  },
+  {
+    name: "Envyfish Egg",
+    available: () =>
+      have($item`envyfish egg`) && get("envyfishMonster") === embezzler && !get("_envyfishEggUsed"),
+    perform: () => {
+      embezzlerPrep();
+      withMacro(Macro.meatKill(), () => use($item`envyfish egg`), true);
+      return get("_envyfishEggUsed");
+    },
+  },
+  {
+    name: "Map for Pills",
+    available: () =>
+      !sober() &&
+      globalOptions.ascending &&
+      clamp(myAdventures() - digitizedMonstersRemaining(), 1, myAdventures()) <=
+        availableAmount($item`Map to Safety Shelter Grimace Prime`),
+    perform: () => {
+      const choiceToSet =
+        availableAmount($item`distention pill`) <
+        availableAmount($item`synthetic dog hair pill`) +
+          availableAmount($item`Map to Safety Shelter Grimace Prime`)
+          ? 1
+          : 2;
+      setChoice(536, choiceToSet);
+      ensureEffect($effect`Transpondent`);
+      use($item`Map to Safety Shelter Grimace Prime`);
+      return true;
+    },
+  },
+  {
+    name: "Barf",
+    available: () => true,
+    perform: () => {
+      useFamiliar(barfFamiliar());
+      const lubing = get("dinseyRollercoasterNext") && have($item`lube-shoes`);
+      meatOutfit(
+        false,
+        lubing ? new Requirement([], { forceEquip: $items`lube-shoes` }) : undefined
+      );
+      adventureMacroAuto(
+        $location`Barf Mountain`,
+        Macro.meatKill(),
+        Macro.if_(
+          `(monsterid ${$monster`Knob Goblin Embezzler`.id}) && !gotjump && !(pastround 2)`,
+          Macro.meatKill()
+        ).abort()
+      );
+      completeBarfQuest();
+      return true;
+    },
+  },
+];
+
+export default function barfTurn(): void {
+  if (SourceTerminal.have()) SourceTerminal.educate([$skill`Extract`, $skill`Digitize`]);
+
+  tryFillLatte();
+  meatMood().execute(estimatedTurns());
+  safeRestore();
```

Comment:
is this dependent on universal recovery or is this just base game junk?

Reply:
We import this from `lib`